gbpservice/nfp/configurator/drivers/firewall/paloalto/paloalto_fw_driver.py

```diff
@@ -0,0 +1,875 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import ast
+import requests
+
+from neutron import context
+from oslo_log import log as logging
+from oslo_serialization import jsonutils
+
+from gbpservice.nfp.configurator.drivers.base import base_driver
+from gbpservice.nfp.configurator.lib import fw_constants as const
+
+import sys
+import json
+import pan.xapi
+import pan.commit
+import pan.config
+import time
+
+LOG = logging.getLogger(__name__)
+
+import sys
+
+
+""" Firewall generic configuration driver for handling device
+configuration requests.
+
+"""
+class FwGenericConfigDriver(object):
+    """
+    Driver class for implementing firewall configuration
+    requests from Orchestrator.
+    """
+
+    def __init__(self):
+        pass
+
+    def _configure_static_ips(self, kwargs):
+        """ Configure static IPs for provider and stitching interfaces
+        of service VM.
+
+        Issues REST call to service VM for configuration of static IPs.
+
+        :param kwargs: a dictionary of firewall rules and objects
+        send by neutron plugin
+
+        Returns: SUCCESS/Failure message with reason.
+
+        """
+        return const.STATUS_SUCCESS
+
+    def _clear_static_ips(self, kwargs):
+        """ Clear static IPs for provider and stitching
+        interfaces of the service VM.
+
+        Issues REST call to service VM for deletion of static IPs.
+
+        :param kwargs: a dictionary of firewall rules and objects
+        send by neutron plugin
+
+        Returns: SUCCESS/Failure message with reason.
+
+        """
+        return const.STATUS_SUCCESS
+
+    def configure_interfaces(self, context, kwargs):
+        """ Configure interfaces for the service VM.
+
+        Calls static IP configuration function and implements
+        persistent rule addition in the service VM.
+        Issues REST call to service VM for configuration of interfaces.
+
+        :param context: neutron context
+        :param kwargs: a dictionary of firewall rules and objects
+        send by neutron plugin
+
+        Returns: SUCCESS/Failure message with reason.
+
+        """
+        time.sleep(const.PA_AUTOCOMMIT_WAITTIME)
```

Comment:
Can this be made deterministic by checking if autocommit is complete or not? Driver get HM call anyways. The HM function for this driver can be implemented in the driver to check HM followed by autocommit complete. Is this possible?